src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -5342,6 +5342,18 @@ static QVariant fcnArrayLast( const QVariantList &values, const QgsExpressionCon
   return list.value( list.size() - 1 );
 }
 
+static QVariant fcnArrayMinimum( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  const QVariantList list = QgsExpressionUtils::getListValue( values.at( 0 ), parent );
+  return list.isEmpty() ? QVariant() : *std::min_element( list.constBegin(), list.constEnd() );
```

Comment:
the QVariant < > operators are deprecated, so you'll need to explicitly handle different types here. You may be able to use qgsVariantLessThan, qgsVariantEqual and qgsVariantGreaterThan here though!

Reply:
Thanks! I added a lamda that uses `qgsVariantLessThan`